src/weewx/almanac.py

```diff
@@ -398,6 +400,10 @@ def get_almanac_data(self, almanac_obj, attr):
                                                context = 'ephem_day',
                                                formatter=almanac_obj.formatter,
                                                converter=almanac_obj.converter)
+        elif attr.startswith('previous_') or attr.startswith('next_'):
+            # Prevent availability checks from being misinterpreted as
+            # heavenly bodies.
+            raise AttributeError(attr)
```

Comment:
The `elif` farther up the file (see line 371 or thereabouts) should have caught this. It checks for specific names (such as `previous_equinox`). If that's not good enough, it could do your check `attr.startswith('previous_')...` etc.

Reply:
This is to prevent tags like `$almanac.previous_aphelion` from being interpreted as a heavenly body's name. The PyEphem almanac does not cover that event, but almanac extensions may add more `previous_...` and `next_...` events. If skin authors use those additional events without checking for the availability of the respective extension this code produces a reasonable error message.